splunk-quarkus/src/main/java/com/redhat/console/integrations/OutgoingCloudEventBuilder.java

```diff
@@ -0,0 +1,59 @@
+package com.redhat.console.integrations;
+
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Message;
+import org.apache.camel.Processor;
+import org.apache.camel.util.json.JsonObject;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+/**
+ * Transformer to create a Map from the outcome of the actual component.
+ */
+/**
+ * Encode the passed body in a CloudEvent, marshalled as Json
+ */
+@ApplicationScoped
+public class OutgoingCloudEventBuilder implements Processor {
+
+    public static final String OUTCOME_EXCHANGE_PROPERTY = "outcome";
+    public static final String SUCCESSFUL_EXCHANGE_PROPERTY = "successful";
+
+    private static final String CE_SPEC_VERSION = "1.0";
+    private static final String CE_TYPE = "com.redhat.console.notifications.history";
+
+    @ConfigProperty(name = "integrations.component.name")
+    String source;
+
+    public void process(Exchange exchange) throws Exception {
+
+        Message in = exchange.getIn();
+
+        JsonObject details = new JsonObject();
+        details.put("target", exchange.getProperty("targetUrl", String.class));
+        details.put("type", in.getHeader("Ce-type", String.class));
+        details.put("outcome", exchange.getProperty(OUTCOME_EXCHANGE_PROPERTY, String.class));
```

Comment:
This PR moves `data.outcome` to `data.details.outcome` on purpose. This change shouldn't break anything in notifications-backend:
https://github.com/RedHatInsights/notifications-backend/blob/f203dc0f1a13445a45130ec8b20671c657258ad1/engine/src/main/java/com/redhat/cloud/notifications/events/CamelHistoryFillerHelper.java#L32-L39